lib/hardware_services/inventory_drivers/MockInventoryDriver.py

```diff
@@ -151,14 +151,47 @@ def load_data(self, quadsinstance, force):
             quadsinstance.logger.error(ex)
             exit(1)
 
-    def write_data(self, quadsinstance, doexit = True):
-        quadsinstance.quads_write_data_(doexit)
+    def write_data(self, quadsinstance, doexit):
+        try:
+            stream = open(quadsinstance.config, 'w')
+            quadsinstance.data = {"clouds":quadsinstance.quads.clouds.data,
+             "hosts":quadsinstance.quads.hosts.data, "history":quadsinstance.quads.history.data, "cloud_history":quadsinstance.quads.cloud_history.data}
+            stream.write( yaml.dump(quadsinstance.data, default_flow_style=False))
+            if doexit:
+                exit(0)
+        except Exception, ex:
+            quadsinstance.logger.error("There was a problem with your file %s" % ex)
+            if doexit:
+                exit(1)
 
     def sync_state(self, quadsinstance):
-        quadsinstance.quads_sync_state_()
+        # sync state
+        if quadsinstance.datearg is not None:
+            quadsinstance.logger.error("--sync and --date are mutually exclusive.")
+            exit(1)
+        for h in sorted(quadsinstance.quads.hosts.data.iterkeys()):
+            default_cloud, current_cloud, current_override = quadsinstance._quads_find_current(h, quadsinstance.datearg)
+            if not os.path.isfile(quadsinstance.statedir + "/" + h):
+                try:
+                    stream = open(quadsinstance.statedir + "/" + h, 'w')
+                    stream.write(current_cloud + '\n')
+                    stream.close()
+                except Exception, ex:
+                    quadsinstance.logger.error("There was a problem with your file %s" % ex)
+        return
 
     def init_data(self, quadsinstance, force):
-        quadsinstance.quads_init_data_(force)
-
+        if not force:
+            if os.path.isfile(quadsinstance.config):
+                quadsinstance.logger.warn("Warning: " + quadsinstance.config + " exists. Use --force to initialize.")
```

Comment:
Since, these if statements are pretty simple, I would suggest we use `and` instead of nested `ifs`.

```python
if not force and os.path.isfile(quadsinstance.config):
    #do stuff
```